nhd/NHDScheduler.py

```diff
@@ -567,8 +607,22 @@ def run(self):
                             if v.active:
                                 self.logger.info(f'Node {item["node"]} already activated for scheduling')
                             else:
-                                active = (self.k8s.IsNodeActive(v.name) and self.ScanAndInitNode(v.name))
+                                # Retry to get node "active" a configurable number of times 
+                                # to accomodate for k8s internal lags when uncordoning node.
+                                for i in range(NODE_ACTIVE_RETRIES):
+                                    active = (self.k8s.IsNodeActive(v.name) and self.ScanAndInitNode(v.name))
+                                    if active:
+                                        break
+                                    self.logger.info(f'Node {v.name} not active yet - will retry status check. Count: {i}')
+                                    time.sleep(1)
```

Comment:
It's probably better to set a wake time here and check in the idle part above. Otherwise you have a single-threaded controller sleeping for up to 10 seconds and causing a backlog of events.